swift-paperless/Views/TLSListView.swift

```diff
@@ -0,0 +1,231 @@
+                    TLSSingleView(identity: identity)
+                } label: {
+                    Text(identity.name)
+                }
+            }
+            .onDelete { ids in
+                withAnimation {
+                    ids.forEach { id in
+                        let item = idenities[id]
+                        Keychain.deleteIdentity(name: item.name)
+                        refreshAll()
+                    }
+                }
+            }
+        }
+        .onAppear {
+            refreshAll()
+        }
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
+                HStack {
+                    NavigationLink(destination: {
+                        CreateView()
+                    }, label: {
+                        Label(String(localized: .localizable(.add)), systemImage: "plus")
+                    })
+                    EditButton()
+                }
+            }
+        }
+        .sheet(isPresented: $showCreate, content: {
+            CreateView()
+                .navigationTitle("T")
+                .toolbar {
+                    ToolbarItem(placement: .navigationBarTrailing) {
+                        HStack {
+                            Button {
+                                showCreate = true
+                                withAnimation {
+                                    // headers.append(.init(key: "Header", value: "Value"))
+                                }
+                            } label: {
+                                Label(String(localized: .localizable(.add)), systemImage: "plus")
+                            }
+                            EditButton()
+                        }
+                    }
+                }
+        })
+    }
+
+    private struct CreateView: View {
+        @State private var certificatePassword: String = ""
+        @State private var certificateName: String = ""
+        @State private var certificateData: Data? = nil
+
+        @State private var isImporting: Bool = false
+        @State private var isCertificateValid = false
+        @State private var certificateState: CertificateState = .notloaded
+
+        @Environment(\.dismiss) var dismiss
+
+        private func validateCertificate(certificateData: Data, certificatePassword: String) -> Bool {
+            do {
+                let _ = try PKCS12(pkcs12Data: certificateData, password: certificatePassword)
+                return true
+            } catch {}
+            return false
+        }
+
+        private func saveToKeychain(certificateData: Data, certificatePassword: String, certificateName: String) {
+            do {
+                let pkc = try PKCS12(pkcs12Data: certificateData, password: certificatePassword)
+                if let identity = pkc.identity {
+                    Keychain.saveIdentity(identity: identity, name: certificateName)
+                }
+            } catch {
+                print(error)
```

Comment:
Use the logging for this. If it can realistically fail, it should probably be surface in the UI.